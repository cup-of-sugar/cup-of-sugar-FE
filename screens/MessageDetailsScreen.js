import * as WebBrowser from "expo-web-browser";
import * as React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Colors from "../constants/Colors";
import { ScrollView } from "react-native-gesture-handler";

export default function MessageDetailsScreen(props) {
  const message = props.route.params.message;

  return (
    <View style={styles.container}>
      <ScrollView
        style={styles.Container}
        scrollIndicatorInsets={{ right: 1 }}
        contentContainerStyle={styles.contentContainer}
      >
        <View style={styles.subjectContainer}>
          <Text style={styles.subject}>Subject: {message.title}</Text>
        </View>
        <View style={styles.messageContainer}>
          <Text style={styles.messageContent}>{message.body}</Text>
        </View>
        <TouchableOpacity
          style={styles.replyButton}
          onPress={() =>
            props.navigation.navigate("Compose", {
              recipient: message.recipient.firstName,
              userId: props.route.params.userId
            })
          }
        >
          <Text style={styles.replyButtonText}>Reply</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    paddingTop: 50,
    height: "100%"
  },
  contentContainer: {
    padding: 15
  },
  messageContainer: {
    borderBottomLeftRadius: 7,
    borderBottomRightRadius: 7,
    borderColor: Colors.lightBlue,
    borderWidth: 3,
    flex: 1
  },
  subjectContainer: {
    backgroundColor: Colors.lightBlue,
    borderTopLeftRadius: 7,
    borderTopRightRadius: 7
  },
  subject: {
    color: "#fff",
    fontSize: 22,
    fontWeight: "bold",
    height: 70,
    padding: 20
  },
  messageContent: {
    fontSize: 20,
    height: 400,
    padding: 15
  },
  replyButton: {
    margin: 40,
    marginTop: 10,
    paddingTop: 20,
    paddingBottom: 20,
    backgroundColor: Colors.lightBlue,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: "#fff"
  },
  replyButtonText: {
    fontSize: 25,
    fontWeight: "bold",
    color: "#fff",
    textAlign: "center"
  }
});
