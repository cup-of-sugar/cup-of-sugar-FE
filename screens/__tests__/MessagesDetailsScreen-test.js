import React from "react";
import MessageDetailsScreen from "../MessageDetailsScreen";
import { MockedProvider } from "@apollo/react-testing";
import ReactTestRenderer from "react-test-renderer";

it("should render without error if on borrowing path", () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: {
      params: {
        action: "borrow",
        userId: "4",
        message: {
          title: "hi",
          body: "how are you",
          recipient: { firstName: "Carole" }
        }
      }
    }
  };
  ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <MessageDetailsScreen {...props} />
    </MockedProvider>
  );
});

it("should render without error if on lending path", () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: {
      params: {
        action: "lend",
        userId: "4",
        message: {
          title: "hi",
          body: "how are you",
          recipient: { firstName: "Carole" }
        }
      }
    }
  };
  ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <MessageDetailsScreen {...props} />
    </MockedProvider>
  );
});
