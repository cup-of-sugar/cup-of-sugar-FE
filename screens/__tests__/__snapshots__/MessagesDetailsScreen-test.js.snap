// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingTop": 50,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 15,
      }
    }
    scrollIndicatorInsets={
      Object {
        "right": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#0C94E1",
            "borderTopLeftRadius": 7,
            "borderTopRightRadius": 7,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 22,
              "fontWeight": "bold",
              "height": 70,
              "padding": 20,
            }
          }
        >
          Subject: Tigers
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomLeftRadius": 7,
            "borderBottomRightRadius": 7,
            "borderColor": "#0C94E1",
            "borderWidth": 3,
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "height": 400,
              "padding": 15,
            }
          }
        >
          When can I get those tigers? Thanks, Carole
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0C94E1",
            "borderColor": "#fff",
            "borderRadius": 10,
            "borderWidth": 1,
            "margin": 40,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 20,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Reply
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
